### DISCUSSION TITLE
Device name not correct

### DISCUSSION THREAD

**Discussion by ulrichwisser** [netalertx/NetAlertX#642] [General]
Hi!
I have a strange error. I use a third level domain for my home network. All devices are named like `device.home.example.com`. When I run `dig -x <ip> PTR` I get the correct name. Bit NetAlertX shows all devices as `device.example.com`. 

What am I doing wrong?

Thankful for any hints

/Ulrich

**Comment by jokob-sk** [comment]
Hi @ulrichwisser ,

That's expected behavior. The app tries to make the device names more user friendly/readable by cleaning up the names:

```
def cleanDeviceName(str, match_IP):
    # alternative str.split('.')[0]
    str = str.replace("._airplay", "")
    str = str.replace("._tcp", "")
    str = str.replace(".localdomain", "")
    str = str.replace(".local", "")    
    str = str.replace("._esphomelib", "")
    str = str.replace("._googlecast", "")
    str = str.replace(".lan", "")
    str = str.replace(".home", "")
    str = re.sub(r'-[a-fA-F0-9]{32}', '', str)    # removing last part of e.g. Nest-Audio-ff77ff77ff77ff77ff77ff77ff77ff77
    str = re.sub(r'#.*', '', str) # Remove everything after '#' including the '#'
    # remove trailing dots
    if str.endswith('.'):
        str = str[:-1]


    if match_IP:
        str = str + " (IP match)"

    return str
```

Let me know if this causes significant issues and I can make this configurable,
Thanks,
j

**Comment by ulrichwisser** [comment]
If I understand right, you try to remove the top level domain if it isn't really usefull like ".local". 
I would argue otherwise, as you can have devices that are discoverable through .local and other that do not use mDNS and are only useable through your their fqdn dns name.

I would propose to change the code to try the following
- try dns reverse and if successfull use that name
- remove all dns labels that start with an "_"
- configurable: remove the dns search list (this would replace .home .lan ...)

Alternatively, if I got something about your current code wrong I would suggest that you remove .home .lan .local .localdomain only if they are used as top level domain, e.g. device.lan -> device, device.lan.home.example.com -> device.lan.home.example.com

**Reply by jokob-sk** [reply]
Hi @ulrichwisser ,

The above method is used in a lot of places so it's not as easy as removing this post-processing completely. 

I'm happy to approve a PR that would improve the above code, be it configurable via a setting on which name parts are removed or otherwise. I'm inclined not to perform and reverse DNS search as names are editable by users and the main properties used to identify a device are the MAC and IP address. 

I had a feature request in the past to remove `local` etc. from names so now there are comp[eting feature requests from different users. 

I'm definitely no network expert, so recommendations on how to handle these conflicting requests are welcome.

**Reply by ulrichwisser** [reply]
Ok, I have a first code version running. But I found another problem. One of my devices showed up with the following name "pi-hole1.no_fqdn_available." Where does that come from?
I'll send you a pull request with my changes. But it is a potentially big change. We can discuss in the pull request.

**Reply by bfellner** [reply]
How about a friendly name and a DNS name field?  I would like this with or without the cleanup :)
